StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs

```diff
@@ -235,28 +235,28 @@ private void HandleDeclaration(SyntaxNodeAnalysisContext context, StyleCopSettin
                     .Where(x => !string.Equals(x.GetName()?.ToString(), XmlCommentHelper.IncludeXmlTag, StringComparison.Ordinal))
                 : documentation.Content.GetXmlElements(this.matchElementName);
 
-            if (!matchingXmlElements.Any())
+            var isInheritingDocumentation =
+                documentation.Content.GetFirstXmlElement(XmlCommentHelper.IncludeXmlTag) is object;
+
+            if (isInheritingDocumentation)
             {
-                var includedDocumentation = documentation.Content.GetFirstXmlElement(XmlCommentHelper.IncludeXmlTag);
-                if (includedDocumentation != null)
+                var declaration = context.SemanticModel.GetDeclaredSymbol(node, context.CancellationToken);
+                var rawDocumentation = declaration?.GetDocumentationCommentXml(expandIncludes: true, cancellationToken: context.CancellationToken);
+                var completeDocumentation = XElement.Parse(rawDocumentation, LoadOptions.None);
+
+                if (this.inheritDocSuppressesWarnings &&
+                    completeDocumentation.Nodes().OfType<XElement>().Any(element => element.Name == XmlCommentHelper.InheritdocXmlTag))
                 {
-                    var declaration = context.SemanticModel.GetDeclaredSymbol(node, context.CancellationToken);
-                    var rawDocumentation = declaration?.GetDocumentationCommentXml(expandIncludes: true, cancellationToken: context.CancellationToken);
-                    var completeDocumentation = XElement.Parse(rawDocumentation, LoadOptions.None);
-
-                    if (this.inheritDocSuppressesWarnings &&
-                        completeDocumentation.Nodes().OfType<XElement>().Any(element => element.Name == XmlCommentHelper.InheritdocXmlTag))
-                    {
-                        // Ignore nodes with an <inheritdoc/> tag in the included XML.
-                        return;
-                    }
-
-                    this.HandleCompleteDocumentation(context, needsComment, completeDocumentation, locations);
-                    return; // done
+                    // Ignore nodes with an <inheritdoc/> tag in the included XML.
+                    return;
                 }
-            }
 
-            this.HandleXmlElement(context, settings, needsComment, matchingXmlElements, locations);
+                this.HandleCompleteDocumentation(context, needsComment, completeDocumentation, locations);
+            }
+            else
+            {
+                this.HandleXmlElement(context, settings, needsComment, matchingXmlElements, locations);
```

Comment:
Small performance optimization: lines 232-236 can be moved here, as matchingXmlElements is only used here

Reply:
Thanks! I have commited the changes to 64de3245af819d2a6c716a919fc47e11c2c24a0a.